lib/backend/syncersv1/updateprocessors/processor.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package updateprocessors
+
+import (
+	"fmt"
+
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/backend/watchersyncer"
+)
+
+// NewGeneralUpdateProcessor implements an update processor that only needs to take in
+// a conversion function. This is only meant to be used in cases where the Process and
+// OnSyncerStarting methods do not have any special handling.
+//
+// One of the main differences between the general update processor and the conflict
+// resolving name cache processor is that the conflict resolving name cache processor
+// handles deletes by inspecting the value in the key-value pair. The general update processor
```

Comment:
Actually the delete - the delete uses the cache to find the old values from an internal cache because it needs the value to construct the v1 key.

Key thing about this "general update processor" is that it's used to provide common processing for v2 resources where the v1 key can be calculated directly from the v2 resource key (i.e. just with the Kind, Name and Namespace).